lib/fog/kubevirt/compute/compute.rb

```diff
@@ -371,40 +356,81 @@ def create_client_from_token(url, version)
               :bearer_token => @kubevirt_token
             }
           }
+          version = detect(url.to_s, @opts[:ssl_options])
+
+          # Return the client immediately if it has been created before:
+          key = url.path + '/' + version
+          client = @clients[key]
+          return client if client
 
-          Kubeclient::Client.new(
+          client = Kubeclient::Client.new(
             url.to_s,
             version,
             @opts
           )
+
+          wrapped_client = ExceptionWrapper.new(client)
+          @clients[key] = wrapped_client
+
+          # Return the client:
+          wrapped_client
         end
 
-        def create_client_from_config(url, version)
+        def create_client_from_config(path)
           config = Kubeclient::Config.read(ENV['KUBECONFIG'] || ENV['HOME']+'/.kube/config')
           context = config.context
+          url = context.api_endpoint
+          version = detect(url + path, context.ssl_options)
 
-          Kubeclient::Client.new(
-            url.to_s,
+          # Return the client immediately if it has been created before:
+          key = path + '/' + version
+          client = @clients[key]
+          return client if client
+
+          client = :Kubeclient::Client.new(
+            url + path,
             version,
             ssl_options: context.ssl_options,
             auth_options: context.auth_options
           )
+
+          wrapped_client = ExceptionWrapper.new(client)
+          @clients[key] = wrapped_client
+
+          # Return the client:
+          wrapped_client
+        end
+
+        def detect(url, ssl_options)
```

Comment:
`detect` tells only half of the story ;-) 
Let's call this one `detectVersion`

Reply:
Ack